setup.py

```diff
@@ -67,9 +67,11 @@ def write_version_file(version):
         f.write("git_version = {}\n".format(repr(sha)))
 
 
-def _get_pytorch_version():
+def _get_pytorch_version(is_nightly):
     # if "PYTORCH_VERSION" in os.environ:
     #     return f"torch=={os.environ['PYTORCH_VERSION']}"
+    if is_nightly:
+        return "torch"
```

Comment:
@skandermoalla I was wondering if we could instruct to install torch nightly (which might be the reason of your problem) but i can't really find how to tell install-requires to use the pip command provided by pytorch...

Reply:
Btw, I was able to build and use a TorchRL wheel by cloning the repo and running `python setup.py bdist_wheel` which uses the local environment instead of building a sandboxed one. I'll add this comment to the issue for reference.

Reply:
And I just noticed that the `torchrl-nightly` was built in an environment using the `torch-nightly`. (https://github.com/pytorch/rl/blob/0afde117bbe11f53e59f89d1d97ff6dd554d3e1b/.github/workflows/nightly_build.yml#L54)
I thought it was using the build dependencies listed in `pyproject.toml` (https://github.com/pytorch/rl/blob/0afde117bbe11f53e59f89d1d97ff6dd554d3e1b/pyproject.toml#L5).

So as you said that's probably why it doesn't work in an environment with non-nightly PyTorch.

Reply:
Maybe we could put `torch>2.1` for the nightlies, which will break with anyone who does not have the nightlies.
I'm not sure how to capture the resulting error and point users in the right direction though as I would imagine that it will simply tell you that this version is not available...

Reply:
I found this (https://stackoverflow.com/questions/24443583/using-an-extra-python-package-index-url-with-setup-py) but it seems deprecated and I'm looking for the equivalent using [`wheel`](https://github.com/pypa/wheel) which provides `bdist_wheel`.

Reply:
Ok let's try with torch 2.1 then, at least there will be an exception thrown
I will have a look at whether we can capture it!

Reply:
This seems to be working, if you want to check.
If you have torch 2.1.0dev installed it will work ok, but if you have not the nightly you'll get
```
ERROR: Could not find a version that satisfies the requirement torch>=2.1.0.dev (from torchrl-nightly) (from versions: 1.7.1, 1.8.0, 1.8.1, 1.9.0, 1.9.1, 1.10.0, 1.10.1, 1.10.2, 1.11.0, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 2.0.0, 2.0.1)
ERROR: No matching distribution found for torch>=2.1.0.dev
```